hat/assets/js/cypress/integration/13 - workflows/list.spec.js

```diff
@@ -0,0 +1,376 @@
+                        interceptFlag = true;
+                        req.reply({
+                            statusCode: 200,
+                            body: {},
+                        });
+                    },
+                ).as('publishVersion');
+                getActionCellButton(2, 2);
+                cy.get('@actionButton').click();
+                cy.get('[data-test="publish-workflow-version"]').should(
+                    'be.visible',
+                );
+
+                cy.get('[data-test="confirm-button"]').click();
+                cy.wait('@publishVersion').then(xhr => {
+                    cy.wrap(xhr.request.body)
+                        .its('status')
+                        .should('eq', 'PUBLISHED');
+                    cy.wrap(interceptFlag).should('eq', true);
+                });
+            });
+
+            it('Unpublish should call api with correct params', () => {
+                interceptFlag = false;
+                cy.intercept(
+                    {
+                        method: 'PATCH',
+                        pathname: '/api/workflowversions/11/',
+                    },
+                    req => {
+                        interceptFlag = true;
+                        req.reply({
+                            statusCode: 200,
+                            body: {},
+                        });
+                    },
+                ).as('unPublishVersion');
+                getActionCellButton(1, 2);
+                cy.get('@actionButton').click();
+                cy.wait('@unPublishVersion').then(xhr => {
+                    cy.wrap(xhr.request.body)
+                        .its('status')
+                        .should('eq', 'UNPUBLISHED');
+                    cy.wrap(interceptFlag).should('eq', true);
+                });
+            });
+
+            it('Delete should open dialog and call api with correct params', () => {
+                interceptFlag = false;
+                cy.intercept(
+                    {
+                        method: 'DELETE',
+                        pathname: '/api/workflowversions/12/',
+                    },
+                    req => {
+                        interceptFlag = true;
+                        req.reply({
+                            statusCode: 200,
+                            body: {},
+                        });
+                    },
+                ).as('deleteVersion');
+                getActionCellButton(0, 1);
+                cy.get('@actionButton').click();
+                cy.get(
+                    '[data-test="delete-dialog-workflow-version-12"]',
+                ).should('be.visible');
+
+                cy.get('[data-test="confirm-button"]').click();
+                cy.wait('@deleteVersion').then(() => {
+                    cy.wrap(interceptFlag).should('eq', true);
+                });
+            });
+            it.skip('Delete ', () => {});
+        });
+    });
```

Comment:
Same here. We could visit the page once and perform the checks in the same order, we don't need to repeat the cy.visit and everything in the `beforeEach`

Reply:
I tried but lookks that buttons are not accessible anymore after first click.... out of the dom